src/app/components/cart-page/cart-page.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { StorageService } from 'src/app/core/services/storage/storage.service';
+
+@Component({
+  selector: 'app-cart-page',
+  templateUrl: './cart-page.component.html',
+  styleUrls: ['./cart-page.component.scss']
+})
+export class CartPageComponent implements OnInit {
+  cartlist: any = [];
+  amount: number = 0;
+  totalAmount: number = 0;
+  private destroy = new Subject<void>();
+
+  constructor(private storage: StorageService) { }
+
+  ngOnInit(): void {
+    this.storage.getProducts()
+      .pipe(takeUntil(this.destroy))
+      .subscribe(data => {
+        this.cartlist = data.filter((p: any) => p.quantity);
```

Comment:
I guess `p` means `products` ? Please name variables appropriately so it clear what it is. Also would be nice if you specified type of `p`.
If it's object with bunch of properties you can create interface for it